src/main/java/org/apache/commons/compress/archivers/zip/ZipUtil.java

```diff
@@ -239,6 +239,7 @@ static void setNameAndCommentFromExtraFields(final ZipArchiveEntry ze,
                                                            originalNameBytes);
         if (newName != null && !originalName.equals(newName)) {
```

Comment:
When I think about it, I would remove the second test in this `if`, so that the Unicode field always have "priority" over the original name as source.
Scenario where it would matter is when the `originalName` contains a non ASCII value (which is properly decoded) and which matches with the Unicode field value.

Reply:
I think the second part is there for a reason, even though my memory is failing me right now - and I stumbled over it myself when I realized the same check is not present for the comment.

I understand what you mean, though, changed with c2906092

Reply:
After some digging through the history it seems the reason no longer exists, I've removed the second part with ced2075c.